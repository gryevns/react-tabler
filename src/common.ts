export type Color =
    | "blue"
    | "azure"
    | "indigo"
    | "purple"
    | "pink"
    | "red"
    | "orange"
    | "yellow"
    | "lime"
    | "green"
    | "teal"
    | "cyan"
    | "gray";

export const colors = [
    "blue" as const,
    "azure" as const,
    "indigo" as const,
    "purple" as const,
    "pink" as const,
    "red" as const,
    "orange" as const,
    "yellow" as const,
    "lime" as const,
    "green" as const,
    "teal" as const,
    "cyan" as const,
    "gray" as const,
];

export type Variant =
    | "primary"
    | "secondary"
    | "success"
    | "info"
    | "warning"
    | "danger";

export const variants = [
    "primary" as const,
    "secondary" as const,
    "success" as const,
    "info" as const,
    "warning" as const,
    "danger" as const,
];

export type Size = "sm" | "md" | "lg" | "xl";

export const sizes = ["sm" as const, "md" as const, "lg" as const, "xl" as const];
