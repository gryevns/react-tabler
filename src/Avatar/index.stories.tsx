import React from "react";

import { colors } from "../common";
import Avatar from "../Avatar";

export default {
    title: "Avatar",
};

const Default = () => (
    <>
        <Avatar>AA</Avatar>
        <Avatar color="red">BB</Avatar>
        <Avatar imageURL="https://picsum.photos/id/128/128">CC</Avatar>
    </>
);

export const Colors = () =>
    colors.map(color => <Avatar color={color} className="mr-1" />);

const sizes = ["sm" as const, "md" as const, "lg" as const, "xl" as const];

export const Sizes = () => sizes.map(size => <Avatar size={size}>{size}</Avatar>);

export const Status = () =>
    colors.map(color => (
        <Avatar status={color} className="mr-1">
            S
        </Avatar>
    ));

const shapes = [
    undefined,
    "rounded" as const,
    "rounded-circle" as const,
    "rounded-0" as const,
    "rounded-lg" as const,
];

export const Shapes = () =>
    shapes.map(shape => (
        <Avatar
            imageURL="https://picsum.photos/id/128/128"
            shape={shape}
            className="mr-1"
        />
    ));

export const List = () => (
    <Avatar.List>
        <Default />
    </Avatar.List>
);

export const Stacked = () => (
    <Avatar.List stacked>
        <Default />
    </Avatar.List>
);
