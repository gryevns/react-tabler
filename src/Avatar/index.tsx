import * as React from "react";
import cn from "classnames";
import { Color, Size } from "../common";

export interface AlertListProps {
    children: React.ReactNode;
    className?: string;
    stacked?: boolean;
}

const AvatarList = ({ children, className, stacked = false }: AlertListProps) => {
    const classes = cn("avatar-list", { "avatar-list-stacked": stacked }, className);
    return <div className={classes}>{children}</div>;
};

export type Shape = "rounded" | "rounded-circle" | "rounded-0" | "rounded-lg";

export interface AvatarProps {
    children?: React.ReactNode;
    className?: string;
    color?: Color;
    imageURL?: string;
    size?: Size;
    shape?: Shape;
    status?: Color;
    style?: React.CSSProperties;
}

export const Avatar = ({
    children,
    className,
    imageURL,
    shape,
    style,
    size,
    status,
    color,
}: AvatarProps) => {
    const classes = cn(
        {
            avatar: true,
            [shape]: true,
            [`avatar-${size}`]: !!size,
            [`bg-${color}-lt`]: !!color,
        },
        className,
    );

    return (
        <span
            className={classes}
            style={
                imageURL ? { ...style, backgroundImage: `url(${imageURL})` } : style
            }
        >
            {status && <span className={`badge bg-${status}`}></span>}
            {children}
        </span>
    );
};

Avatar.List = AvatarList;

export default Avatar;
