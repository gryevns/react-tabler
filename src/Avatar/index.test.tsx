import React from "react";
import { shallow } from "enzyme";

import { colors, Color, sizes, Size } from "../common";
import Avatar, { Shape } from "./index";

describe("Avatar", () => {
    it("should match color snapshot", () => {
        colors.map((color: Color) => {
            const wrapper = shallow(<Avatar color={color} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
    it("should match size snapshot", () => {
        sizes.map((size: Size) => {
            const wrapper = shallow(<Avatar size={size} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
    it("should match status snapshot", () => {
        colors.map((color: Color) => {
            const wrapper = shallow(<Avatar status={color} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
    it("should match shape snapshot", () => {
        const shapes = ["", "rounded", "rounded-circle", "rounded-0", "rounded-lg"];
        shapes.map((shape: Shape) => {
            const wrapper = shallow(<Avatar shape={shape} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
    it("should match list snapshot", () => {
        const wrapper = shallow(
            <Avatar.List>
                <Avatar>1</Avatar>
                <Avatar>2</Avatar>
            </Avatar.List>,
        );
        expect(wrapper).toMatchSnapshot();
    });
    it("should match stacked snapshot", () => {
        const wrapper = shallow(
            <Avatar.List stacked>
                <Avatar>1</Avatar>
                <Avatar>2</Avatar>
            </Avatar.List>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
