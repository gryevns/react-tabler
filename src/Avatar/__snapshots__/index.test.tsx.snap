// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should match color snapshot 1`] = `
<span
  className="avatar undefined bg-blue-lt"
/>
`;

exports[`Avatar should match color snapshot 2`] = `
<span
  className="avatar undefined bg-azure-lt"
/>
`;

exports[`Avatar should match color snapshot 3`] = `
<span
  className="avatar undefined bg-indigo-lt"
/>
`;

exports[`Avatar should match color snapshot 4`] = `
<span
  className="avatar undefined bg-purple-lt"
/>
`;

exports[`Avatar should match color snapshot 5`] = `
<span
  className="avatar undefined bg-pink-lt"
/>
`;

exports[`Avatar should match color snapshot 6`] = `
<span
  className="avatar undefined bg-red-lt"
/>
`;

exports[`Avatar should match color snapshot 7`] = `
<span
  className="avatar undefined bg-orange-lt"
/>
`;

exports[`Avatar should match color snapshot 8`] = `
<span
  className="avatar undefined bg-yellow-lt"
/>
`;

exports[`Avatar should match color snapshot 9`] = `
<span
  className="avatar undefined bg-lime-lt"
/>
`;

exports[`Avatar should match color snapshot 10`] = `
<span
  className="avatar undefined bg-green-lt"
/>
`;

exports[`Avatar should match color snapshot 11`] = `
<span
  className="avatar undefined bg-teal-lt"
/>
`;

exports[`Avatar should match color snapshot 12`] = `
<span
  className="avatar undefined bg-cyan-lt"
/>
`;

exports[`Avatar should match color snapshot 13`] = `
<span
  className="avatar undefined bg-gray-lt"
/>
`;

exports[`Avatar should match list snapshot 1`] = `
<div
  className="avatar-list"
>
  <Component>
    1
  </Component>
  <Component>
    2
  </Component>
</div>
`;

exports[`Avatar should match shape snapshot 1`] = `
<span
  className="avatar "
/>
`;

exports[`Avatar should match shape snapshot 2`] = `
<span
  className="avatar rounded"
/>
`;

exports[`Avatar should match shape snapshot 3`] = `
<span
  className="avatar rounded-circle"
/>
`;

exports[`Avatar should match shape snapshot 4`] = `
<span
  className="avatar rounded-0"
/>
`;

exports[`Avatar should match shape snapshot 5`] = `
<span
  className="avatar rounded-lg"
/>
`;

exports[`Avatar should match size snapshot 1`] = `
<span
  className="avatar undefined avatar-sm"
/>
`;

exports[`Avatar should match size snapshot 2`] = `
<span
  className="avatar undefined avatar-md"
/>
`;

exports[`Avatar should match size snapshot 3`] = `
<span
  className="avatar undefined avatar-lg"
/>
`;

exports[`Avatar should match size snapshot 4`] = `
<span
  className="avatar undefined avatar-xl"
/>
`;

exports[`Avatar should match stacked snapshot 1`] = `
<div
  className="avatar-list avatar-list-stacked"
>
  <Component>
    1
  </Component>
  <Component>
    2
  </Component>
</div>
`;

exports[`Avatar should match status snapshot 1`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-blue"
  />
</span>
`;

exports[`Avatar should match status snapshot 2`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-azure"
  />
</span>
`;

exports[`Avatar should match status snapshot 3`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-indigo"
  />
</span>
`;

exports[`Avatar should match status snapshot 4`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-purple"
  />
</span>
`;

exports[`Avatar should match status snapshot 5`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-pink"
  />
</span>
`;

exports[`Avatar should match status snapshot 6`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-red"
  />
</span>
`;

exports[`Avatar should match status snapshot 7`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-orange"
  />
</span>
`;

exports[`Avatar should match status snapshot 8`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-yellow"
  />
</span>
`;

exports[`Avatar should match status snapshot 9`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-lime"
  />
</span>
`;

exports[`Avatar should match status snapshot 10`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-green"
  />
</span>
`;

exports[`Avatar should match status snapshot 11`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-teal"
  />
</span>
`;

exports[`Avatar should match status snapshot 12`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-cyan"
  />
</span>
`;

exports[`Avatar should match status snapshot 13`] = `
<span
  className="avatar undefined"
>
  <span
    className="badge bg-gray"
  />
</span>
`;
